api/internal/core/store/store.go

```diff
@@ -54,7 +54,7 @@ type GenericStore struct {
 type GenericStoreOption struct {
 	BasePath   string
 	ObjType    reflect.Type
-	KeyFunc    func(obj interface{}) string
+	KeyFunc    func(obj interface{}, key string) string
```

Comment:
Wow, this is not a good way, `KeyFunc` is responsible to ensure resource key, not the place to do compatible logic.
We can do it in `StringToObjPtr`.

Reply:
updated.